godot-kotlin/godot-library/src/nativeMain/kotlin/godot/annotation/PropertyTypeHintAnnotation.kt

```diff
@@ -0,0 +1,266 @@
+package godot.annotation
+
+import godot.core.GodotResource
+import godot.registration.Range
+import kotlin.reflect.KClass
+
+/*
+All type checks will happen at the entry-generator at the moment.
+We should move that to the idea plugin at some point in time to provide IDE help and not just compilation errors.
+ */
+
+/**
+ * Can only be used on Int properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class IntRange(val start: Int, val end: Int, val step: Int = -1, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on Float properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class FloatRange(val start: Float, val end: Float, val step: Float = -1f, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class DoubleRange(val start: Double, val end: Double, val step: Double = -1.0, val or: Range = Range.NONE)
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ExpRange(val start: Float, val end: Float, val step: Float = -1f)
+
+/**
+ * This property hint is optional!
+ *
+ * A Enum that is visible to the editor is always marked as PROPERTY_HINT_ENUM if not defined otherwise
+ * through another PropertyTypeHintAnnotation. It then will enumerate using strings.
+ *
+ * You only have to use this annotation if you want to enumerate enums with ints in the editor.
+ *
+ * Can only be used on Enum properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class EnumTypeHint(val enumerateAsInt: Boolean = false)
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ExpEasing(val attenuation: Boolean = false, val inout: Boolean = true)
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Lenght(val lenght: Int = -1)
+
+/**
+ * Can only be used on Map<Enum, Boolean> properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Flags
+
+/**
+ * Can only be used on Int properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Layers2DPhysics
```

Comment:
Not in same order as `object.h` in engine:
```cpp
	PROPERTY_HINT_LAYERS_2D_RENDER,
	PROPERTY_HINT_LAYERS_2D_PHYSICS,
	PROPERTY_HINT_LAYERS_3D_RENDER,
	PROPERTY_HINT_LAYERS_3D_PHYSICS,
```
I let you choose if you merge with that order or not.

Reply:
oh not seen this sorry. I will cahnge it